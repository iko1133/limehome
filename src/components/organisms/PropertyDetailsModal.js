import React from "react";
import { StyleSheet } from "react-native";
import styled from "styled-components/native";

import Button from "atoms/Button";
import CloseButton from "atoms/CloseButton";
import Divider from "atoms/Divider";
import IconText from "atoms/IconText";
import ImagesCarousel from "molecules/ImagesCarousel";
import Tag from "atoms/Tag";

const Modal = styled.Modal``;

const InnerContainer = styled.ScrollView`
  background-color: ${({ theme }) => theme.BACKGROUND_COLOR};
`;

const ModalCloseButton = styled(CloseButton)`
  left: 12px;
  top: 12px;
  position: absolute;
`;

const DetailsContainer = styled.View`
  padding: 20px 12px;
`;

const TopInfoContainer = styled.View`
  flex-direction: row;
`;

const TopLeftInfoContainer = styled.View`
  flex: 1;
`;

const Name = styled.Text`
  font-size: 20px;
  font-weight: bold;
`;

const PropertyLocationText = styled(IconText)``;

const PropertyRatingText = styled(IconText)`
  border: 1px solid ${({ theme }) => theme.DEFAULT_BORDER_COLOR};
  border-radius: 6px;
  padding: 6px;
`;

const Description = styled.Text`
  margin: 12px 0;
  line-height: 24px;
  font-size: 15px;
`;

const AvailableOptionsLabel = styled.Text`
  font-size: 18px;
  font-weight: bold;
  margin: 4px 0 12px 0;
`;

const TagsContainer = styled.View`
  margin: 0 -2px;
  flex-direction: row;
  flex-wrap: wrap;
`;

const OptionsTag = styled(Tag)`
  margin: 2px;
`;

const Footer = styled.View`
  align-items: center;
  background-color: ${({ theme }) => theme.DARKER_BACKGROUND_COLOR};
  flex-direction: row;
  justify-content: space-between;
  padding: 8px 16px;
  width: 100%;
`;

const PropertyPriceFromLabel = styled.Text``;

const PropertyPriceText = styled.Text`
  color: ${({ theme }) => theme.SPECIAL_TEXT_COLOR};
`;

/**
 * A modal showing a the information of the property given
 * @param {boolean} visible - a boolean, indicating whether the modal is visible or not
 * @param {object} propertyDetails - an object containing the information of the property displayed by the component
 * @param {function} toggleModalVisibility - a function to toggle modal visibility
 */

const PropertyDetailsModal = ({
  visible,
  propertyDetails = {},
  toggleModalVisibility,
}) => {
  return (
    <Modal visible={visible}>
      <InnerContainer showsVerticalScrollIndicator={false}>
        {visible && <ImagesCarousel images={propertyDetails.images} />}
        <ModalCloseButton onPress={toggleModalVisibility} />

        <DetailsContainer>
          <TopInfoContainer>
            <TopLeftInfoContainer>
              <Name>{propertyDetails.name}</Name>

              <PropertyLocationText
                text={`${Math.round(
                  propertyDetails.distance
                )} km from city center`}
                iconName="location-pin"
              />
            </TopLeftInfoContainer>

            <PropertyRatingText
              text="4.5"
              iconName="star"
              iconSize={26}
              iconLeft={false}
              textStyle={styles.propertyRatingTextStyle}
            />
          </TopInfoContainer>

          <Description>{propertyDetails.description}</Description>

          <Divider />

          <AvailableOptionsLabel>
            Room types available in this location
          </AvailableOptionsLabel>

          <TagsContainer>
            {propertyDetails.unit_groups?.map((group) => (
              <OptionsTag key={`tag_${group.id}`} text={group.title} />
            ))}
          </TagsContainer>
        </DetailsContainer>
      </InnerContainer>

      <Footer>
        <PropertyPriceFromLabel>
          From{" "}
          <PropertyPriceText>
            {(propertyDetails.lowest_price_per_night ?? 0).toFixed(2)}
            â‚¬/Night
          </PropertyPriceText>
        </PropertyPriceFromLabel>

        <Button
          text="EXPLORE"
          iconLeft={false}
          iconName="arrow-forward"
          textStyle={styles.exploreButtonTextStyle}
          iconColor="white"
        />
      </Footer>
    </Modal>
  );
};

const styles = StyleSheet.create({
  propertyRatingTextStyle: { fontSize: 20, fontWeight: "bold" },
  exploreButtonTextStyle: { color: "white" },
});

export default PropertyDetailsModal;
