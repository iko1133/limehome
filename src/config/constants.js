export const MAP_DEFAULT_COORDINATES = {
  latitude: 52.520008,
  longitude: 13.404954,
  latitudeDelta: 0.15,
  longitudeDelta: 0.15,
};
