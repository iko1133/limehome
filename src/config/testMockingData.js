export const PROPERTIES = [
  {
    id: 99,
    name: "berlin schlüterstraße",
    city: "berlin",
    city_id: 32,
    street: "schlüterstraße",
    location: {
      lat: 52.50176279999999,
      lng: 13.3164893,
      city: "Berlin",
    },
    distance: 6.321157219990815,
    images: [
      {
        url: "https://limehome.imgix.net/properties/99/9fee5516-9d41-41f7-95ea-6ac72c49fb86.jpg",
        is_portrait: false,
        position: 0,
        unit_group_ids: [16],
        tags: [],
      },
    ],
    lowest_price_per_night: null,
    lowest_price_per_month: null,
  },
  {
    id: 152,
    name: "berlin werftstraße",
    city: "berlin",
    city_id: 32,
    street: "werftstraße ",
    location: {
      lat: 52.52259309999999,
      lng: 13.3589213,
      city: "Berlin",
    },
    distance: 3.127749150912429,
    images: [
      {
        url: "https://limehome.imgix.net/properties/152/bff791de-57fa-450b-869d-0a7bfac7a55c.jpg",
        is_portrait: false,
        position: 0,
        unit_group_ids: [178],
        tags: [],
      },
    ],
    lowest_price_per_night: null,
    lowest_price_per_month: null,
  },
];

export const FIXED_PROPERTIES = [
  {
    id: 99,
    name: "berlin schlüterstraße",
    city: "berlin",
    city_id: 32,
    street: "schlüterstraße",
    location: {
      latitude: 52.50176279999999,
      longitude: 13.3164893,
      city: "Berlin",
    },
    distance: 6.321157219990815,
    images: [
      {
        url: "https://limehome.imgix.net/properties/99/9fee5516-9d41-41f7-95ea-6ac72c49fb86.jpg",
        is_portrait: false,
        position: 0,
        unit_group_ids: [16],
        tags: [],
      },
    ],
    lowest_price_per_night: null,
    lowest_price_per_month: null,
  },
  {
    id: 152,
    name: "berlin werftstraße",
    city: "berlin",
    city_id: 32,
    street: "werftstraße ",
    location: {
      latitude: 52.52259309999999,
      longitude: 13.3589213,
      city: "Berlin",
    },
    distance: 3.127749150912429,
    images: [
      {
        url: "https://limehome.imgix.net/properties/152/bff791de-57fa-450b-869d-0a7bfac7a55c.jpg",
        is_portrait: false,
        position: 0,
        unit_group_ids: [178],
        tags: [],
      },
    ],
    lowest_price_per_night: null,
    lowest_price_per_month: null,
  },
];
