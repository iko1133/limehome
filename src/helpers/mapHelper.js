// Copies given object, except, switches names of the location property's properties for the
// marker objects to function properly
// lgn to longitude
// lat to latitude
export const fixObjectLocationProperty = (object) => ({
  ...object,
  location: {
    ...object.location,
    longitude: object.location?.lng,
    latitude: object.location?.lat,
  },
});
