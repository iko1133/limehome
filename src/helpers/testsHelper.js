import { ThemeProvider } from "styled-components/native";

import { lightTheme } from "styles/theme";

/**
 * Wraps given children into a ThemeProvider with lightTheme set as the main theme
 * @param {object} children - children to be rendered inside ThemeProvider
 */
export const WrapInThemeProvider = (children) => (
  <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
);
