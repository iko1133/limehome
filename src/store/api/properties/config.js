// list of urls needed
const config = {
  propertiesList: "properties/v1/public/properties/",
  propertyDetails: "properties/v1/public/properties/",
};

export default config;
