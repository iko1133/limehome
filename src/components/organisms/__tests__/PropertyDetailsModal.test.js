import React from "react";
import renderer from "react-test-renderer";

import { WrapInThemeProvider } from "helpers/testsHelper";
import { FIXED_PROPERTIES } from "config/testMockingData";
import PropertyDetailsModal from "../PropertyDetailsModal";

// Fixing vector icons problem with jest
jest.mock(
  "@expo/vector-icons/build/vendor/react-native-vector-icons/lib/create-icon-set.js",
  () => {
    return () => "";
  }
);

describe("Testing PropertyDetailsModal", () => {
  it("Renders property details modal with property 0 on it", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(
          <PropertyDetailsModal propertyDetails={FIXED_PROPERTIES[0]} />
        )
      )
    ).toMatchSnapshot());

  it("Renders property details modal with property 1 on it", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(
          <PropertyDetailsModal propertyDetails={FIXED_PROPERTIES[1]} />
        )
      )
    ).toMatchSnapshot());

  it("Renders property details modal nonvisible", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(<PropertyDetailsModal visible={false} />)
      )
    ).toMatchSnapshot());

  it("Renders property details modal nonvisible, but with propertyDetails defined", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(
          <PropertyDetailsModal
            visible={false}
            propertyDetails={FIXED_PROPERTIES[1]}
          />
        )
      )
    ).toMatchSnapshot());
});
