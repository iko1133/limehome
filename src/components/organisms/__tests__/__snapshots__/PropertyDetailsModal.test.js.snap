// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PropertyDetailsModal Renders property details modal nonvisible 1`] = `
<Modal
  hardwareAccelerated={false}
  style={
    Array [
      Object {},
    ]
  }
  visible={false}
>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
    style={
      Array [
        Object {
          "backgroundColor": "#F7F1E9",
        },
      ]
    }
  >
    <View>
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Array [
            Object {},
            Object {
              "left": 12,
              "position": "absolute",
              "top": 12,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F7F1E9",
              "borderRadius": 6,
              "height": 35,
              "justifyContent": "center",
              "opacity": 1,
              "width": 35,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                },
              ]
            }
          >
            X
          </Text>
        </View>
      </RCTSafeAreaView>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 20,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 20,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {},
                ]
              }
            >
              <Text />
              <Text
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                NaN km from city center
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "borderColor": "#9D9E9F",
                  "borderRadius": 6,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "paddingBottom": 6,
                  "paddingLeft": 6,
                  "paddingRight": 6,
                  "paddingTop": 6,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {},
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              4.5
            </Text>
            <Text />
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "fontSize": 15,
                "lineHeight": 24,
                "marginBottom": 12,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 12,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "backgroundColor": "darkgray",
                "height": 1,
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 8,
                "width": "100%",
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 12,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 4,
              },
            ]
          }
        >
          Room types available in this location
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "marginBottom": 0,
                "marginLeft": -2,
                "marginRight": -2,
                "marginTop": 0,
              },
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#CACBAB",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
        ]
      }
    >
      From
       
      <Text
        style={
          Array [
            Object {
              "color": "#B26422",
            },
          ]
        }
      >
        0.00
        €/Night
      </Text>
    </Text>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#4D6447",
          "borderRadius": 4,
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {},
              Object {
                "color": "white",
              },
            ]
          }
        >
          EXPLORE
        </Text>
        <Text />
      </View>
    </View>
  </View>
</Modal>
`;

exports[`Testing PropertyDetailsModal Renders property details modal nonvisible, but with propertyDetails defined 1`] = `
<Modal
  hardwareAccelerated={false}
  style={
    Array [
      Object {},
    ]
  }
  visible={false}
>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
    style={
      Array [
        Object {
          "backgroundColor": "#F7F1E9",
        },
      ]
    }
  >
    <View>
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Array [
            Object {},
            Object {
              "left": 12,
              "position": "absolute",
              "top": 12,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F7F1E9",
              "borderRadius": 6,
              "height": 35,
              "justifyContent": "center",
              "opacity": 1,
              "width": 35,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                },
              ]
            }
          >
            X
          </Text>
        </View>
      </RCTSafeAreaView>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 20,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 20,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              berlin werftstraße
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {},
                ]
              }
            >
              <Text />
              <Text
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                3 km from city center
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "borderColor": "#9D9E9F",
                  "borderRadius": 6,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "paddingBottom": 6,
                  "paddingLeft": 6,
                  "paddingRight": 6,
                  "paddingTop": 6,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {},
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              4.5
            </Text>
            <Text />
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "fontSize": 15,
                "lineHeight": 24,
                "marginBottom": 12,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 12,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "backgroundColor": "darkgray",
                "height": 1,
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 8,
                "width": "100%",
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 12,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 4,
              },
            ]
          }
        >
          Room types available in this location
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "marginBottom": 0,
                "marginLeft": -2,
                "marginRight": -2,
                "marginTop": 0,
              },
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#CACBAB",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
        ]
      }
    >
      From
       
      <Text
        style={
          Array [
            Object {
              "color": "#B26422",
            },
          ]
        }
      >
        0.00
        €/Night
      </Text>
    </Text>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#4D6447",
          "borderRadius": 4,
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {},
              Object {
                "color": "white",
              },
            ]
          }
        >
          EXPLORE
        </Text>
        <Text />
      </View>
    </View>
  </View>
</Modal>
`;

exports[`Testing PropertyDetailsModal Renders property details modal with property 0 on it 1`] = `
<Modal
  hardwareAccelerated={false}
  style={
    Array [
      Object {},
    ]
  }
  visible={true}
>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
    style={
      Array [
        Object {
          "backgroundColor": "#F7F1E9",
        },
      ]
    }
  >
    <View>
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Array [
            Object {},
            Object {
              "left": 12,
              "position": "absolute",
              "top": 12,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F7F1E9",
              "borderRadius": 6,
              "height": 35,
              "justifyContent": "center",
              "opacity": 1,
              "width": 35,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                },
              ]
            }
          >
            X
          </Text>
        </View>
      </RCTSafeAreaView>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 20,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 20,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              berlin schlüterstraße
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {},
                ]
              }
            >
              <Text />
              <Text
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                6 km from city center
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "borderColor": "#9D9E9F",
                  "borderRadius": 6,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "paddingBottom": 6,
                  "paddingLeft": 6,
                  "paddingRight": 6,
                  "paddingTop": 6,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {},
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              4.5
            </Text>
            <Text />
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "fontSize": 15,
                "lineHeight": 24,
                "marginBottom": 12,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 12,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "backgroundColor": "darkgray",
                "height": 1,
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 8,
                "width": "100%",
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 12,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 4,
              },
            ]
          }
        >
          Room types available in this location
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "marginBottom": 0,
                "marginLeft": -2,
                "marginRight": -2,
                "marginTop": 0,
              },
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#CACBAB",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
        ]
      }
    >
      From
       
      <Text
        style={
          Array [
            Object {
              "color": "#B26422",
            },
          ]
        }
      >
        0.00
        €/Night
      </Text>
    </Text>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#4D6447",
          "borderRadius": 4,
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {},
              Object {
                "color": "white",
              },
            ]
          }
        >
          EXPLORE
        </Text>
        <Text />
      </View>
    </View>
  </View>
</Modal>
`;

exports[`Testing PropertyDetailsModal Renders property details modal with property 1 on it 1`] = `
<Modal
  hardwareAccelerated={false}
  style={
    Array [
      Object {},
    ]
  }
  visible={true}
>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
    style={
      Array [
        Object {
          "backgroundColor": "#F7F1E9",
        },
      ]
    }
  >
    <View>
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Array [
            Object {},
            Object {
              "left": 12,
              "position": "absolute",
              "top": 12,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F7F1E9",
              "borderRadius": 6,
              "height": 35,
              "justifyContent": "center",
              "opacity": 1,
              "width": 35,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                },
              ]
            }
          >
            X
          </Text>
        </View>
      </RCTSafeAreaView>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 20,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 20,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              berlin werftstraße
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {},
                ]
              }
            >
              <Text />
              <Text
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                3 km from city center
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "borderColor": "#9D9E9F",
                  "borderRadius": 6,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "paddingBottom": 6,
                  "paddingLeft": 6,
                  "paddingRight": 6,
                  "paddingTop": 6,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {},
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              4.5
            </Text>
            <Text />
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "fontSize": 15,
                "lineHeight": 24,
                "marginBottom": 12,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 12,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "backgroundColor": "darkgray",
                "height": 1,
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 8,
                "width": "100%",
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 12,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 4,
              },
            ]
          }
        >
          Room types available in this location
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "marginBottom": 0,
                "marginLeft": -2,
                "marginRight": -2,
                "marginTop": 0,
              },
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#CACBAB",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
        ]
      }
    >
      From
       
      <Text
        style={
          Array [
            Object {
              "color": "#B26422",
            },
          ]
        }
      >
        0.00
        €/Night
      </Text>
    </Text>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#4D6447",
          "borderRadius": 4,
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {},
              Object {
                "color": "white",
              },
            ]
          }
        >
          EXPLORE
        </Text>
        <Text />
      </View>
    </View>
  </View>
</Modal>
`;
